pandas/core/indexes/range.py

```diff
@@ -23,9 +23,11 @@
 class RangeIndex(Int64Index):
 
     """
-    Immutable Index implementing a monotonic range. RangeIndex is a
-    memory-saving special case of Int64Index limited to representing
-    monotonic ranges.
+    Immutable Index implementing a monotonic integer range.
+
+    RangeIndex is a memory-saving special case of Int64Index limited to
+    representing monotonic ranges. This is the default index type used
+    by DataFrame and Series when no explicit index is provided by the user.
```

Comment:
Same as previous `RangeIndex` comment: could potentially mention that it makes some index operations computationally efficient.

Reply:
Do you have any examples, so I can understand it?

Reply:
Since `RangeIndex` is defined entirely from the `start`,` stop`, and `step`, you can use these values to simply compute some properties in constant time instead of needing to do a pass over the index in linear time, e.g. `max`, `min`.  Similarly, operations between two `RangeIndex` can be computed in constant time, e.g. `intersection`, `equals`.  You also get some properties for free (or constant time) that can be beneficial for index operations, e.g. monotonic, non-null, uniqueness.

Reply:
I can see it can be fasterst index type (also simply by virtue of not having to build a Int64-index everytime a df is created), but I feel more complex advantages should be referenced from the doc string to an explanation elsewere rather than being explained in the doc string itself. I.e. the doc string should only make that statement if there's some further explanation elsewere, e.g. in the user guide (IMO), which there isn't ATM (AFAICT).

Maybe you could take a stab at explaining this succintly in the user guide and add a later PR that builds on top of mine?

Reply:
I didn't mean to suggest that you add details about it in depth, just merely mention the fact.  Like how you're not actually explaining why `RangeIndex` is memory-saving, you just mention that it is.  So just something along the lines of "memory-efficient" -> "memory and computationally efficient" was all I was suggesting.